javalib/src/main/scala/java/io/DataOutputStream.scala

```diff
@@ -46,32 +83,41 @@ class DataOutputStream(out: OutputStream)
     writeInt(java.lang.Float.floatToIntBits(v))
 
   override final def writeInt(v: Int): Unit = {
-    write((v >> 24) & 0xFF)
-    write((v >> 16) & 0xFF)
-    write((v >> 8) & 0xFF)
-    write(v & 0xFF)
+    byteArray(0) = (v >> 24).toByte
+    byteArray(1) = (v >> 16).toByte
+    byteArray(2) = (v >> 8).toByte
+    byteArray(3) = (v & 0xFF).toByte
+    write(byteArray, 0, 4)
   }
 
   override final def writeLong(v: Long): Unit = {
-    write(((v >> 56) & 0xFF).toInt)
-    write(((v >> 48) & 0xFF).toInt)
-    write(((v >> 40) & 0xFF).toInt)
-    write(((v >> 32) & 0xFF).toInt)
-    write(((v >> 24) & 0xFF).toInt)
-    write(((v >> 16) & 0xFF).toInt)
-    write(((v >> 8) & 0xFF).toInt)
-    write((v & 0xFF).toInt)
+    byteArray(0) = (v >> 56).toByte
+    byteArray(1) = (v >> 48).toByte
+    byteArray(2) = (v >> 40).toByte
+    byteArray(3) = (v >> 32).toByte
+    byteArray(4) = (v >> 24).toByte
+    byteArray(5) = (v >> 16).toByte
+    byteArray(6) = (v >> 8).toByte
+    byteArray(7) = (v & 0xFF).toByte
+    write(byteArray, 0, 8)
   }
 
   override final def writeShort(v: Int): Unit = {
-    write((v >> 8) & 0xFF)
-    write(v & 0xFF)
+    byteArray(0) = (v >> 8).toByte
+    byteArray(1) = (v & 0xFF).toByte
+    write(byteArray, 0, 2)
   }
 
+  // Optimizing total memory usage in toModifiedUTF() and its
+  // private methods is left as an exercise for the reader.
```

Comment:
"an exercise for the reader" makes no sense in the code. The reader can't do anything about it.

Consider a more factual description, for example:

> The current implementation is not optimized for total memory usage.

Reply:
Gone.  I had been hoping to encourage the reader to submit a PR to bring about
the next evolution of the code.